### PR TITLE
#200 Debugging contracts

### LINKED ISSUE #200: Document debugging possibility
Make mention of printing in hints, compiling with hint validation disabled, using integrated-devnet stdout.
Consider this for reference: https://github.com/Shard-Labs/starknet-devnet/#contract-debugging
Also potentially find where this was explained in the Telegram starknet core stars group

### PR DESCRIPTION
**Pull request by shramee**
## Checklist:

-   [ ] Formatted the code
-   [ ] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Documented the changes
    -   [ ] If `src/type-extensions.ts` was changed, updated the line number referencing this file in the `## API` section of README.md
-   [ ] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.ts`)
-   [ ] Linked issues which this PR resolves
-   [ ] Created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example):
    -   < EXAMPLE_REPO_PR_URL > 
    -   [ ] Modified `test.sh` to use my example repo branch
    -   [ ] Restored `test.sh` to to use the original branch (after the example repo PR has been merged)
-   [ ] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

Closes #200

### CODE REVIEW

File: www/docs/intro.md [hunk 1]
```diff
@@ -375,6 +376,43 @@ Devnet offers [empty block creation](https://shard-labs.github.io/starknet-devne
...
+
+DevNet logs look like this,
+
+```
+Adding balance...
+0
+25
+```
+
+When using `integrated-devnet` you can specify `stdout` as `STDOUT` or a file in your hardhat config file. Documented in [runtime network](#runtime-network---integrated-devnet) section.
```
Comment (FabijanC) [inline www/docs/intro.md:414]: Here I would motivate the adoption of integrated-devnet by saying:
```
If you want to have your debug lines printed in the same terminal as your hardhat script/test, use integrated-devnet with `stdout` set to `"STDOUT"` as documented in ...
```

**Review by FabijanC — CHANGES_REQUESTED** [review]
Thanks for the good example! I would just improve the integrated-devnet section a bit.

File: www/docs/intro.md [hunk 2]
```diff
@@ -375,6 +376,43 @@ Devnet offers [empty block creation](https://shard-labs.github.io/starknet-devne
...
+    %{ print( "Adding balance..." ) %}
+    %{ print( ids.res ) %}
+    balance.write(res + amount);
+    let (afterUpdate) = balance.read();
+    %{ print( ids.afterUpdate ) %}
+    return ();
+}
+```
+
+DevNet logs look like this,
```
Comment (FabijanC) [inline www/docs/intro.md:406]: Also let's keep the naming uniform and use `Devnet` instead of `DevNet`

Comment (shramee) [reply www/docs/intro.md]: Addressed in dd2e5e617cd5241599ab5bde9729ebaea8252f1b

File: www/docs/intro.md (hunk 1 above)
Comment (shramee) [reply www/docs/intro.md]: Addressed in dd2e5e617cd5241599ab5bde9729ebaea8252f1b

**Comment by shramee** [comment]
Sorry guys @Nathan-SL and @FabijanC,
The reviewer switch was unintentional, just clicked the request review refresh icon after updating but it changed the reviewer. 😰

**Review by Nathan-SL — APPROVED** [review]

**Pull request merged**